Credentials/credentials_student.php

```diff
@@ -25,134 +32,94 @@
+            echo "<div class='error'>" . $e->getMessage() . '</div>';
         }
 
-        if ($result->rowCount() != 1) {
+        if ($students->getResultCount() != 1) {
             echo "<div class='error'>";
             echo __('The selected record does not exist, or you do not have access to it.');
             echo '</div>';
         } else {
-            $row = $result->fetch();
+            $student = $students->getRow(0);
 
             $page->breadcrumbs->add(__('Manage Credentials'), 'credentials.php', [
                 'search' => $search,
                 'allStudents' => $allStudents,
             ]);
-            $page->breadcrumbs->add(formatName('', $row['preferredName'], $row['surname'], 'Student'));
+            $page->breadcrumbs->add(Format::name('', $student['preferredName'], $student['surname'], 'Student'));
 
             if (isset($_GET['return'])) {
                 returnProcess($guid, $_GET['return'], null, null);
             }
 
-            try {
-                $data = array('gibbonPersonID' => $gibbonPersonID);
-                $sql = 'SELECT *
-                    FROM credentialsCredential
-                        JOIN credentialsWebsite ON (credentialsCredential.credentialsWebsiteID=credentialsWebsite.credentialsWebsiteID)
-                    WHERE gibbonPersonID=:gibbonPersonID
-                        AND active=\'Y\'
-                    ORDER BY title';
-                $result = $connection2->prepare($sql);
-                $result->execute($data);
-            } catch (PDOException $e) {
-                echo "<div class='error'>".$e->getMessage().'</div>';
-            }
-
             if ($search != '' or $allStudents != '') {
                 echo "<div class='linkTop'>";
-                echo "<a href='".$_SESSION[$guid]['absoluteURL']."/index.php?q=/modules/Credentials/credentials.php&search=$search&allStudents=$allStudents'>".__('Back to Search Results').'</a>';
+                echo "<a href='" . $_SESSION[$guid]['absoluteURL'] . "/index.php?q=/modules/Credentials/credentials.php&search=$search&allStudents=$allStudents'>" . __('Back to Search Results') . '</a>';
                 echo '</div>';
             }
 
             echo "<div class='linkTop'>";
-            echo "<a href='".$_SESSION[$guid]['absoluteURL'].'/index.php?q=/modules/'.$_SESSION[$guid]['module']."/credentials_student_add.php&gibbonPersonID=$gibbonPersonID&search=$search&allStudents=$allStudents'>".__('Add')."<img style='margin-left: 5px' title='".__('Add')."' src='./themes/".$_SESSION[$guid]['gibbonThemeName']."/img/page_new.png'/></a>";
+            echo "<a href='" . $_SESSION[$guid]['absoluteURL'] . '/index.php?q=/modules/' . $_SESSION[$guid]['module'] . "/credentials_student_add.php&gibbonPersonID=$gibbonPersonID&search=$search&allStudents=$allStudents'>" . __('Add') . "<img style='margin-left: 5px' title='" . __('Add') . "' src='./themes/" . $_SESSION[$guid]['gibbonThemeName'] . "/img/page_new.png'/></a>";
```

Comment:
DataTables have an addHeaderAction method, which helps remove these bits of HTML, and takes care of passing values to the action buttons. Eg:

```php
$table->addHeaderAction('add', __('Add'))
      ->addParam('gibbonPersonID', $gibbonPersonID)
      ->addParam('search', $search)
      ->addParam('allStudents', $allStudents)
      ->setURL('/modules/Credentials/credentials_student_add.php')
      ->displayLabel();
```